### PR TITLE
Filter controller watch events based on namespace

### PR DESCRIPTION
**Pull request by mbaijal**
### What does this PR do / how does this improve the operators?
When the namespace is provided, the operator only manages events from that namespace.

### Which issue(s) does this PR fix?

Fixes #

### Special notes for the reviewer:
I have tested this using a locally built controller image. It does not break existing functionality. 

### Does this PR require changes to documentation?
I will follow this up with another PR which adds the updated helm charts and that will contain a README on how to use namespaces. 

### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you written or refactored unit tests to cover the change?
* [ ] Have you ran all unit tests and ensured they are passing?
* [ ] Have you manually tested each feature that is being added/modified?
* [ ] Have you ensured you have not introduced linting errors?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -50,9 +50,11 @@ func init() {
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
+	var namespace string
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&namespace, "namespace", "", "The namespace that the manager's cache can watch objects in.")
```
Comment (RedbackThomson) [inline main.go]: nit: This wording sounds a bit ambiguous. Maybe `The namespace in which the manager controls and reconciles resources`?

Comment (mbaijal) [reply main.go]: I actually picked it up from some k8s documentation i think

Comment (RedbackThomson) [reply main.go]: It's not even grammatically correct haha. That's awkward

Comment (mbaijal) [reply main.go]: Fixed.

File: main.go [hunk 2]
```diff
@@ -61,12 +63,15 @@ func main() {
...
 		MetricsBindAddress: metricsAddr,
 		LeaderElection:     enableLeaderElection,
+		Namespace:          namespace,
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
+	ctrl.Log.WithName("Namespace").Info(namespace)
```
Comment (gautamkmr) [inline main.go]: I think we can say something like `starting manager in namespace <namespace name> ` or just merge this with error message in line 69.

Comment (cadedaniel) [reply main.go]: +1 to improving error message (if we merge with line 69 it won't show up in success cases).

File: main.go [hunk 3]
```diff
@@ -50,9 +50,11 @@ func init() {
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
+	var namespace string
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&namespace, "namespace", "", "The namespace in which the manager controls and reconciles resources.")
```
Comment (cadedaniel) [inline main.go]: Maybe add "Leave blank for watch all resources."?

File: main.go (hunk 2 above)
Comment (mbaijal) [reply main.go]: Fair, will update the message. I think it is good to have a log even in case there is no error.

File: main.go (hunk 3 above)
Comment (mbaijal) [reply main.go]: done

File: main.go (hunk 2 above)
Comment (mbaijal) [reply main.go]: done

**Pull request merged**